src/test/java/uk/gov/hmcts/reform/slc/services/servicebus/MessageProcessorTest.java

```diff
@@ -91,4 +98,86 @@ public void should_not_call_action_if_there_are_no_messages_on_queue() throws Ex
         verify(sendLetterService, never()).send(any());
         verifyNoMoreInteractions(insights);
     }
+
+    @Test
+    public void should_not_process_message_when_connection_exception_is_thrown_on_acquiring_sb_queue_connection() {
+        //given
+        processor = new MessageProcessor(receiverProvider, sendLetterService);
+        ReflectionTestUtils.setField(processor, "insights", insights);
+
+        doThrow(ConnectionException.class).when(receiverProvider).get();
+
+        //when
+        Throwable exception = catchThrowable(() -> {
+            processor.process();
+        });
+
+        //then
+        assertThat(exception).isNull();
+
+        verify(insights).trackException(any(ConnectionException.class));
+        verify(sendLetterService, never()).send(any());
+        verifyNoMoreInteractions(insights);
+    }
+
+    @Test
+    public void should_not_process_message_when_servicebus_exception_is_thrown_on_closing_sb_queue_connection() {
+        //given
+        processor = new MessageProcessor(receiverProvider, sendLetterService);
+        ReflectionTestUtils.setField(processor, "insights", insights);
+
+        doThrow(ServiceBusException.class).when(receiverProvider).get();
```

Comment:
test method says: 'on_closing_sb_queue_connection'

Reply:
I guess we're closing the queue after processing the message, anyway.